x-pack/plugins/security_solution/public/exceptions/pages/shared_lists/index.tsx

```diff
@@ -393,6 +394,44 @@ export const SharedLists = React.memo(() => {
     setDisplayAddExceptionItemFlyout(false);
     setIsCreatePopoverOpen(false);
   };
+  const onCreateExceptionListOpenClick = () => setDisplayCreateSharedListFlyout(true);
+
+  const isReadOnly = useMemo(() => {
+    return (canUserREAD && !canUserCRUD) ?? true;
+  }, [canUserREAD, canUserCRUD]);
+
+  useEffect(() => {
```

Comment:
In this component, we have quite hard async logic with 3 different loading indicators

`loading`, `loadingExceptions`, `loadingTableInfo` (depends on `loadingExceptions`)

It wasn't easy to understand this code initially, maybe it will be easy if we extract some logic condition to variables like that? 

I did not test this code, but idea should be there
```js
const exceptionsLoaded = !loadingExceptions && !loadingTableInfo && !initLoading;
const hasNoExceptions = exceptionsLoaded && !exceptionListsWithRuleRefs.length;
const isSearchingExceptions = viewerStatus === ViewerStatus.SEARCHING;
const isLoadingExceptions = viewerStatus === ViewerStatus.LOADING;


if (isSearchingExceptions && hasNoExceptions) {
    setViewStatus(ViewerStatus.EMPTY_SEARCH);
} else if (!exceptionsLoaded) {
    setViewStatus(ViewerStatus.LOADING);
} else if (isLoadingExceptions && hasNoExceptions) {
    setViewStatus(ViewerStatus.EMPTY);
} else if (isLoadingExceptions && exceptionsLoaded) {
    setViewStatus(null);
}
```

Reply:
Updated! Looks much better.